lucene/core/src/java/org/apache/lucene/index/IndexWriterConfig.java

```diff
@@ -484,6 +487,23 @@ public IndexWriterConfig setCommitOnClose(boolean commitOnClose) {
     return this;
   }
 
+  /**
+   * Expert: sets the amount of time to wait for merges returned by MergePolicy.findCommitMerges(...).
+   * If this time is reached, we proceed with the commit based on segments merged up to that point.
```

Comment:
Maybe also say that any merges still running after this time will still run to completion, so they are not wasted.

Reply:
Done